### ISSUE TITLE
`test_parallel_gates_equivalence_groups()` fails with new Cirq update

### ISSUE THREAD

**Issue by vtomole** [SupertechLabs/cirq-superstaq#210]
https://github.com/SupertechLabs/cirq-superstaq/runs/5683094136?check_suite_focus=true

**Comment by vtomole** [comment]
Apprently

```python
import cirq
import cirq_superstaq

res = cirq_superstaq.ParallelGates(cirq.X, cirq_superstaq.ZZSwapGate(1.23), cirq.X).on(cirq.LineQubit(0), cirq.LineQubit(1), cirq.LineQubit(2), cirq.LineQubit(3))
res1 = cirq_superstaq.ParallelGates(cirq.X, cirq_superstaq.ZZSwapGate(1.23), cirq.X).on(cirq.LineQubit(3), cirq.LineQubit(1), cirq.LineQubit(2), cirq.LineQubit(0))

print(res)
print(res1)
print(res == res1)
```

Should be `True`, but now it's `False`

**Comment by vtomole** [comment]
Taking this as @richrines1 is busy.

**Comment by vtomole** [comment]
We've merged https://github.com/SupertechLabs/cirq-superstaq/pull/211 temporarily until this issue is closed.

**Comment by richrines1** [comment]
this issue was fixed in cirq 0.15. We added a workaround so that tests still pass using cirq 0.14.x (#237), but technically the issue still exists for users using that version

that said imo we can close this because (1) it only comes up in really niche cases, (2) the issue is on cirq's end so there doesn't seem to be an easy workaround, and (3) even if there was we'll probably update our requirements to `cirq>=0.15.0` before we get around to fixing it. @vtomole?
